.github/workflows/erlang-parallel-build.yml

```
@@ -0,0 +1,224 @@
+            ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-build-
+
+      - name: Check formatting
+        run: rebar3 fmt -c
+
+      - name: Run linting
+        run: rebar3 lint
+
+      - name: Run xref
+        run: rebar3 xref
+
+  dialyze:
+    name: Dialyze
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup BEAM
+        uses: erlef/setup-beam@v1.10
+        with:
+          otp-version: ${{ inputs.otp-version }}
+          rebar3-version: ${{ inputs.rebar-version }}
+
+      - name: Cache _build
+        uses: actions/cache@v2
+        with:
+          path: _build/*/lib
+          key: ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-build-${{ hashFiles('rebar.lock') }}
+          restore-keys: |
+            ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-build-
+
+      - name: Cache PLTs
+        uses: actions/cache@v2
+        with:
+          path: _build/test/rebar3_*_plt
+          key: ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-plt-${{ hashFiles('rebar.lock') }}
+          restore-keys: |
+            ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-plt-      
+
+      - name: Run dialyzer
+        run: rebar3 as test dialyzer
+
+  test:
+    name: Test
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup BEAM
+        uses: erlef/setup-beam@v1.10
+        with:
+          otp-version: ${{ inputs.otp-version }}
+          rebar3-version: ${{ inputs.rebar-version }}
+
+      - name: Cache _build
+        uses: actions/cache@v2
+        with:
+          path: _build/*/lib
+          key: ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-build-${{ hashFiles('rebar.lock') }}
+          restore-keys: |
+            ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-build-
+
+      - name: Run EUnit
+        run: rebar3 eunit --cover
+
+      - name: Run CommonTest
+        id: run-ct
+        if: ${{ inputs.run-ct-with-compose == false }}
+        run: rebar3 ct --cover
+
+      - name: Run CommonTest (/w docker-compose)
+        id: run-ct-w-compose
+        if: ${{ inputs.run-ct-with-compose == true }}
+        env:
+          # Pass workflow params to use in docker-compose.yml
+          DEV_IMAGE_TAG: ${{ inputs.service-name }}-dev
+          OTP_VERSION: ${{ inputs.otp-version }}
+          THRIFT_VERSION: ${{ inputs.thrift-version }}
+          # Enable buildkit extensions in docker compose
+          COMPOSE_DOCKER_CLI_BUILD: true
+          DOCKER_BUILDKIT: true
+        run: |
+          [[ "${{ inputs.service-name }}" == "" ]] && exit 1
+          docker-compose run --use-aliases --rm ${{ inputs.service-name }} rebar3 ct --cover
```

Пока такое предложение смог сформулировать.

Вводим конвенцию, что `run-ct-compose-container-name` по умолчанию `testrunner` или что-то вроде того, и стараемся этой конвенции везде следовать.

```suggestion
          docker-compose run --use-aliases --rm ${{ inputs.run-ct-compose-container-name }} rebar3 ct --cover
```

Соответственно, в `docker-compose.yml` этот контейнер тоже называем всегда `testrunner`. На том же уровне возможно понадобиться (ну либо по вкусу) добавить `hostname` / `aliases` для этого контейнера.

Что скажете?

---

Заодно избавляемся от постыдного коммента про `Keep in sync with ...`.

---

Тоже посещала какая-то такая мысль, думаю норм